lib/collibra/client/get_rights_to_use_asset.ts

```diff
@@ -1,29 +1,38 @@
-import { RequesterFn } from "./make_collibra_client"
+import { AxiosResponse } from "axios"
+import * as E from "fp-ts/Either"
+import * as TE from "fp-ts/TaskEither"
+import { pipe } from "fp-ts/lib/function"
 
-export const getRightsToUseAsset: RequesterFn<Collibra.Asset> = (client) => async (id: string) => {
-  const {
-    data: {
-      results: [relationType],
-    },
-  } = await client.get<Collibra.PagedRelationTypeResponse>(`/relationTypes`, {
-    params: {
-      sourceTypeName: "data product",
-      targetTypeName: "rights-to-use",
-    },
-  })
+type RelationTypesResponse = AxiosResponse<Collibra.PagedRelationTypeResponse>
+type RelationsResponse = AxiosResponse<Collibra.PagedRelationResponse>
+type AssetResponse = AxiosResponse<Collibra.Asset>
 
-  const {
-    data: {
-      results: [relation],
-    },
-  } = await client.get<Collibra.PagedRelationResponse>(`/relations`, {
-    params: {
-      relationTypeId: relationType.id,
-      sourceId: id,
-    },
-  })
-
-  const { data: asset } = await client.get<Collibra.Asset>(`/assets/${relation.target.id}`)
-
-  return asset
-}
+export const getRightsToUseAsset = (client: Net.Client) => (id: string) =>
+  pipe(
+    TE.tryCatch(
+      () =>
```

Comment:
One bonus change that I might consider is to pull these lambdas into their own functions so that you could write something like `TE.tryCatch(getRelationTypes)`

Reply:
not a bad idea